## Initial Data Collection and Cleaning
In this notebook we will:
1. Use two different datasets from Kaggle that took game information from steam.com 
2. We will do slight cleaning on each dataset then merge them to make one
3. Further cleaning will be done on the combined dataset 
4. This dataset will then be used for out recommendation system

***
## Import neccessary packages:

In [1]:
import warnings
warnings.filterwarnings('ignore')
import pandas as pd
from collections import Counter 
from itertools import dropwhile
import matplotlib.pyplot as plt
import pickle

*** 
## Data Collecting and Cleaning:
This is where we will collect and clean the datasets

In [2]:
steam = pd.read_csv('data/steam.csv')

steam_games = pd.read_csv('data/steam_games.csv')

### First dataset
Start by looking at the first data set we called in and understand what information we have.

In [3]:
steam.head()

appid                       name release_date  english         developer  \
0     10             Counter-Strike   2000-11-01        1             Valve   
1     20      Team Fortress Classic   1999-04-01        1             Valve   
2     30              Day of Defeat   2003-05-01        1             Valve   
3     40         Deathmatch Classic   2001-06-01        1             Valve   
4     50  Half-Life: Opposing Force   1999-11-01        1  Gearbox Software   

  publisher          platforms  required_age  \
0     Valve  windows;mac;linux             0   
1     Valve  windows;mac;linux             0   
2     Valve  windows;mac;linux             0   
3     Valve  windows;mac;linux             0   
4     Valve  windows;mac;linux             0   

                                          categories  genres  \
0  Multi-player;Online Multi-Player;Local Multi-P...  Action   
1  Multi-player;Online Multi-Player;Local Multi-P...  Action   
2              Multi-player;Valve Anti-Cheat enabled  Action   
3  Multi-player;Online Multi-Player;Local Multi-P...  Action   
4  Single-player;Multi-player;Valve Anti-Cheat en...  Action   

                  steamspy_tags  achievements  positive_ratings  \
0        Action;FPS;Multiplayer             0            124534   
1        Action;FPS;Multiplayer             0              3318   
2  FPS;World War II;Multiplayer             0              3416   
3        Action;FPS;Multiplayer             0              1273   
4             FPS;Action;Sci-fi             0              5250   

   negative_ratings  average_playtime  median_playtime             owners  \
0              3339             17612              317  10000000-20000000   
1               633               277               62   5000000-10000000   
2               398               187               34   5000000-10000000   
3               267               258              184   5000000-10000000   
4               288               624              415   5000000-10000000   

   price  
0   7.19  
1   3.99  
2   3.99  
3   3.99  
4   3.99

In [4]:
steam.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27075 entries, 0 to 27074
Data columns (total 18 columns):
appid               27075 non-null int64
name                27075 non-null object
release_date        27075 non-null object
english             27075 non-null int64
developer           27075 non-null object
publisher           27075 non-null object
platforms           27075 non-null object
required_age        27075 non-null int64
categories          27075 non-null object
genres              27075 non-null object
steamspy_tags       27075 non-null object
achievements        27075 non-null int64
positive_ratings    27075 non-null int64
negative_ratings    27075 non-null int64
average_playtime    27075 non-null int64
median_playtime     27075 non-null int64
owners              27075 non-null object
price               27075 non-null float64
dtypes: float64(1), int64(8), object(9)
memory usage: 3.7+ MB


#### steam dataframe descriptions 
|column name | description |
| --- | ---|
|appid| Unique identifier for each title|
|name| Title of app (game)|
|release_date | Release date in format YYYY-MM-DD|
|english | Language support: 1 if is in English| 
|developer| Name (or names) of developer(s). Semicolon delimited if multiple|
|publisher | Name (or names) of publisher(s). Semicolon delimited if multiple |
|platforms | Semicolon delimited list of supported platforms. At most includes: windows;mac;linux| 
|required_age | Minimum required age according to PEGI UK standards. Many with 0 are unrated or unsupplied|
|categories| Semicolon delimited list of game categories, e.g. single-player; multi-player|
|genres|Semicolon delimited list of game genres, e.g. action;adventure|
|steamspy_tags| Semicolon delimited list of top steamspy game tags, similar to genres but community voted, e.g. action;adventure|
|achievements|Number of in-games achievements, if any|
|positive_ratings|Number of positive ratings, from SteamSpy|
|negative_ratings|Number of negative ratings, from SteamSpy|
|average_playtime|Average user playtime, from SteamSpy|
|median_playtime|Median user playtime, from SteamSpy|
|owners|Estimated number of owners. Contains lower and upper bound (like 20000-50000)|
|price|Current full price of title in GBP, (pounds sterling)|


### Second dataset
Next look at the second data set we called in and understand what information we have there.

In [5]:
steam_games.head()

url types  \
0    https://store.steampowered.com/app/379720/DOOM/   app   
1  https://store.steampowered.com/app/578080/PLAY...   app   
2  https://store.steampowered.com/app/637090/BATT...   app   
3    https://store.steampowered.com/app/221100/DayZ/   app   
4  https://store.steampowered.com/app/8500/EVE_On...   app   

                            name  \
0                           DOOM   
1  PLAYERUNKNOWN'S BATTLEGROUNDS   
2                     BATTLETECH   
3                           DayZ   
4                     EVE Online   

                                        desc_snippet  \
0  Now includes all three premium DLC packs (Unto...   
1  PLAYERUNKNOWN'S BATTLEGROUNDS is a battle roya...   
2  Take command of your own mercenary outfit of '...   
3  The post-soviet country of Chernarus is struck...   
4  EVE Online is a community-driven spaceship MMO...   

                                      recent_reviews  \
0  Very Positive,(554),- 89% of the 554 user revi...   
1  Mixed,(6,214),- 49% of the 6,214 user reviews ...   
2  Mixed,(166),- 54% of the 166 user reviews in t...   
3  Mixed,(932),- 57% of the 932 user reviews in t...   
4  Mixed,(287),- 54% of the 287 user reviews in t...   

                                         all_reviews  release_date  \
0  Very Positive,(42,550),- 92% of the 42,550 use...  May 12, 2016   
1  Mixed,(836,608),- 49% of the 836,608 user revi...  Dec 21, 2017   
2  Mostly Positive,(7,030),- 71% of the 7,030 use...  Apr 24, 2018   
3  Mixed,(167,115),- 61% of the 167,115 user revi...  Dec 13, 2018   
4  Mostly Positive,(11,481),- 74% of the 11,481 u...   May 6, 2003   

             developer                                publisher  \
0          id Software    Bethesda Softworks,Bethesda Softworks   
1     PUBG Corporation        PUBG Corporation,PUBG Corporation   
2  Harebrained Schemes  Paradox Interactive,Paradox Interactive   
3  Bohemia Interactive  Bohemia Interactive,Bohemia Interactive   
4                  CCP                                  CCP,CCP   

                                        popular_tags  \
0  FPS,Gore,Action,Demons,Shooter,First-Person,Gr...   
1  Survival,Shooter,Multiplayer,Battle Royale,PvP...   
2  Mechs,Strategy,Turn-Based,Turn-Based Tactics,S...   
3  Survival,Zombies,Open World,Multiplayer,PvP,Ma...   
4  Space,Massively Multiplayer,Sci-fi,Sandbox,MMO...   

                                        game_details  \
0  Single-player,Multi-player,Co-op,Steam Achieve...   
1             Multi-player,Online Multi-Player,Stats   
2  Single-player,Multi-player,Online Multi-Player...   
3  Multi-player,Online Multi-Player,Steam Worksho...   
4  Multi-player,Online Multi-Player,MMO,Co-op,Onl...   

                                           languages  achievements  \
0  English,French,Italian,German,Spanish - Spain,...          54.0   
1  English,Korean,Simplified Chinese,French,Germa...          37.0   
2                      English,French,German,Russian         128.0   
3  English,French,Italian,German,Spanish - Spain,...           NaN   
4                      English,German,Russian,French           NaN   

                                               genre  \
0                                             Action   
1             Action,Adventure,Massively Multiplayer   
2                          Action,Adventure,Strategy   
3             Action,Adventure,Massively Multiplayer   
4  Action,Free to Play,Massively Multiplayer,RPG,...   

                                    game_description  \
0   About This Game Developed by id software, the...   
1   About This Game  PLAYERUNKNOWN'S BATTLEGROUND...   
2   About This Game  From original BATTLETECH/Mec...   
3   About This Game The post-soviet country of Ch...   
4                                  About This Game     

                                      mature_content  \
0                                                NaN   
1   Mature Content Description  The developers de...   
2                       

In [6]:
steam_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40833 entries, 0 to 40832
Data columns (total 20 columns):
url                         40833 non-null object
types                       40831 non-null object
name                        40817 non-null object
desc_snippet                27612 non-null object
recent_reviews              2706 non-null object
all_reviews                 28470 non-null object
release_date                37654 non-null object
developer                   40490 non-null object
publisher                   35733 non-null object
popular_tags                37888 non-null object
game_details                40313 non-null object
languages                   40797 non-null object
achievements                12194 non-null float64
genre                       40395 non-null object
game_description            37920 non-null object
mature_content              2897 non-null object
minimum_requirements        21069 non-null object
recommended_requirements    21075 non-null

#### steam_games dataframe descriptions 
|column name | description |
| --- | ---|
|url| Url of a game|
|types|type of package - app, sub or bundle|
|name|Name of a game|
|desc_snippet|short description of a game|
|recent_reviews|recent reviews|
|all_reviews|all reviews|
|release_date|release date|
|developer|developer of a game|
|publisher|publisher or publishers of a game|
|popular_tags|tags|
|game_details|details of a game|
|languages|supported languages|
|achievements|number of achievements|
|genre|genre(s) of a game|
|game_description|game description|
|mature_content|description of mature content in a game|
|minimum_requirements|minimum specs for a game|
|recommended_requirements|recommended specs for a game|
|original_price|price without discount|
|discount_price|price with discount|


### Merge and Clean
Now we will merge the two data sets by the game name and do further cleaning.

In [7]:
steam['name'] = steam['name'].str.replace(r'[^\w\s]+', '').str.strip().str.lower()
steam_games['name'] = steam_games['name'].str.replace(r'[^\w\s]+', '').str.strip().str.lower()

df = steam.merge(steam_games, left_on='name', right_on='name')

In [8]:
df.head()

appid                     name release_date_x  english       developer_x  \
0     10            counterstrike     2000-11-01        1             Valve   
1     20    team fortress classic     1999-04-01        1             Valve   
2     30            day of defeat     2003-05-01        1             Valve   
3     40       deathmatch classic     2001-06-01        1             Valve   
4     50  halflife opposing force     1999-11-01        1  Gearbox Software   

  publisher_x          platforms  required_age  \
0       Valve  windows;mac;linux             0   
1       Valve  windows;mac;linux             0   
2       Valve  windows;mac;linux             0   
3       Valve  windows;mac;linux             0   
4       Valve  windows;mac;linux             0   

                                          categories  genres  ...  \
0  Multi-player;Online Multi-Player;Local Multi-P...  Action  ...   
1  Multi-player;Online Multi-Player;Local Multi-P...  Action  ...   
2              Multi-player;Valve Anti-Cheat enabled  Action  ...   
3  Multi-player;Online Multi-Player;Local Multi-P...  Action  ...   
4  Single-player;Multi-player;Valve Anti-Cheat en...  Action  ...   

                                        game_details  \
0  Multi-player,Online Multi-Player,Local Multi-P...   
1  Multi-player,Online Multi-Player,Local Multi-P...   
2              Multi-player,Valve Anti-Cheat enabled   
3  Multi-player,Online Multi-Player,Local Multi-P...   
4  Single-player,Multi-player,Valve Anti-Cheat en...   

                                           languages  achievements_y   genre  \
0  English,French,German,Italian,Spanish - Spain,...             NaN  Action   
1  English,French,German,Italian,Spanish - Spain,...             NaN  Action   
2      English,French,German,Italian,Spanish - Spain             NaN  Action   
3  English,French,German,Italian,Spanish - Spain,...             NaN  Action   
4                       English,French,German,Korean             NaN  Action   

                                    game_description  \
0   About This Game Play the world's number 1 onl...   
1   About This Game One of the most popular onlin...   
2   About This Game Enlist in an intense brand of...   
3   About This Game Enjoy fast-paced multiplayer ...   
4   About This Game Return to the Black Mesa Rese...   

                                      mature_content minimum_requirements  \
0   Mature Content Description  The developers de...                  NaN   
1   Mature Content Description  The developers de...                  NaN   
2                                                NaN                  NaN   
3                                                NaN                  NaN   
4                                                NaN                  NaN   

   recommended_requirements original_price discount_price  
0                       NaN          $9.99         $65.70  
1                       NaN          $4.99         $65.70  
2                       NaN          $4.99         $65.70  
3                       NaN          $4.99         $65.70  
4                       NaN          $4.99         $65.70  

[5 rows x 37 columns]

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 19747 entries, 0 to 19746
Data columns (total 37 columns):
appid                       19747 non-null int64
name                        19747 non-null object
release_date_x              19747 non-null object
english                     19747 non-null int64
developer_x                 19747 non-null object
publisher_x                 19747 non-null object
platforms                   19747 non-null object
required_age                19747 non-null int64
categories                  19747 non-null object
genres                      19747 non-null object
steamspy_tags               19747 non-null object
achievements_x              19747 non-null int64
positive_ratings            19747 non-null int64
negative_ratings            19747 non-null int64
average_playtime            19747 non-null int64
median_playtime             19747 non-null int64
owners                      19747 non-null object
price                       19747 non-null float6

#### To drop from df: 

|column name | info | why |
| --- | --- | --- |
|`appid`|`19744 non-null int64`|no need|
|`english`|`19744 non-null int64`|no need|
|`url`|`19744 non-null object`|no need|
|`recent_reviews`|`2465 non-null object`|not enough|
|`release_date_y`|`19734 non-null object`|doubled with `release_date_x`|
|`developer_y`|`19739 non-null object`|doubled with `developer_x`|
|`publisher_y`|`19733 non-null object`|doubled with `publisher_x`|
|`game_details`|`19739 non-null object`|same as `categories`|
|`achievements_y`|`11347 non-null float64`|doubled with `achievements_x`|
|`genre`|`19737 non-null object`|doubled with `genres`|
|`game_description`|`19734 non-null object`|same as `desc_snippet`|
|`minimum_requirements`|`10370 non-null object`|not enough|
|`recommended_requirements`|`10375 non-null object`|not enough|
|`original_price`|`19659 non-null object`|doubled with `price`|
|`discount_price`|`6458 non-null object`|not enough|

We are also going to drop the columns on reviews and play times as we only care about number of sales which is on the nunber of owners. These columns are:

|column name | info |
| --- | --- |
|`positive_ratings`|`19747 non-null int64`|
|`negative_ratings`|`19747 non-null int64`|
|`all_reviews`|`9120 non-null object`|
|`average_playtime`|`19747 non-null int64`|
|`median_playtime`|`19747 non-null int64`|

In [10]:
to_drop = ['appid', 'english', 'url', 'recent_reviews', 'release_date_y', 'developer_y', 'publisher_y',
           'game_details', 'achievements_y', 'genre', 'game_description','steamspy_tags', 
           'minimum_requirements', 'recommended_requirements', 'original_price', 'discount_price',
          'positive_ratings', 'negative_ratings', 'all_reviews', 'average_playtime', 'median_playtime']

df = df.drop(to_drop, axis = 1) 

df = df.rename(columns={'release_date_x': 'release_date', 'developer_x': 'developer', 'publisher_x' : 'publisher',
                  'achievements_x' : 'achievements'})

In [11]:
df.drop(df[df['price'] == 0].index, inplace = True) 

In [12]:
df = df.fillna(value = {'mature_content': 'none'})

df['owners'] = df['owners'].str.split('-').apply(lambda x: (int(x[0]) + int(x[1])) / 2).astype(int)

df['est_revenue'] = df['owners'] * df['price']

df['year'] = pd.DatetimeIndex(df['release_date']).year
df['month'] = pd.DatetimeIndex(df['release_date']).month

import calendar
df['month'] = df['month'].apply(lambda x: calendar.month_abbr[x])
df['month'] = df['month'].str.lower()

In [13]:
spring = ['mar', 'apr', 'may']
for i in spring:
    df.loc[df['month'].str.contains(i), 'season'] = 'spring'
summer = ['jun', 'jul', 'aug']
for i in summer:
    df.loc[df['month'].str.contains(i), 'season'] = 'summer'
fall = ['sep', 'oct','nov']
for i in fall:
    df.loc[df['month'].str.contains(i), 'season'] = 'fall'
winter = ['jan', 'feb', 'dec']
for i in winter:
    df.loc[df['month'].str.contains(i), 'season'] = 'winter'

In [14]:
df = df.dropna()

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 17930 entries, 0 to 19746
Data columns (total 20 columns):
name              17930 non-null object
release_date      17930 non-null object
developer         17930 non-null object
publisher         17930 non-null object
platforms         17930 non-null object
required_age      17930 non-null int64
categories        17930 non-null object
genres            17930 non-null object
achievements      17930 non-null int64
owners            17930 non-null int64
price             17930 non-null float64
types             17930 non-null object
desc_snippet      17930 non-null object
popular_tags      17930 non-null object
languages         17930 non-null object
mature_content    17930 non-null object
est_revenue       17930 non-null float64
year              17930 non-null int64
month             17930 non-null object
season            17930 non-null object
dtypes: float64(2), int64(4), object(14)
memory usage: 2.9+ MB


#### df dataframe descriptions after cleaning 
|column name | description |
| --- | ---|
|name| Title of app (game)|
|release_date| Release date in format YYYY-MM-DD|
|developer| Name (or names) of developer(s). Semicolon delimited if multiple|
|publisher| Name (or names) of publisher(s). Semicolon delimited if multiple |
|platforms | Semicolon delimited list of supported platforms. At most includes: windows;mac;linux| 
|required_age | Minimum required age according to PEGI UK standards. Many with 0 are unrated or unsupplied|
|categories| Semicolon delimited list of game categories, e.g. single-player; multi-player|
|genres|Semicolon delimited list of game genres, e.g. action;adventure|
|steamspy_tags| Semicolon delimited list of top steamspy game tags, similar to genres but community voted, e.g. action;adventure|
|achievements|Number of in-games achievements, if any|
|positive_ratings|Number of positive ratings, from SteamSpy|
|negative_ratings|Number of negative ratings, from SteamSpy|
|average_playtime|Average user playtime, from SteamSpy|
|median_playtime|Median user playtime, from SteamSpy|
|owners|Estimated owners|
|price|Current full price of title in GBP, (pounds sterling)|
|types|type of package - app, sub or bundle|
|name|Name of a game|
|desc_snippet|short description of a game|
|all_reviews|all reviews|
|popular_tags|tags|
|languages|supported languages|
|est_revenue| estimated revenue (owners * price)|
|year|year game was released|
|month|month game was released|
|season|season game was released|

In [16]:
df['developer'] = df['developer'].str.strip().str.lower()
df['publisher'] = df['publisher'].str.strip().str.lower()
df['types'] = df['types'].str.strip().str.lower()
df['mature_content'] = df['mature_content'].str.strip().str.lower()

df['platforms'] = df['platforms'].str.strip().str.lower().str.replace(' ','').str.split(';').str.join(' ')
df['genres'] = df['genres'].str.strip().str.lower().str.replace(' ','').str.split(';').str.join(' ')
df['categories'] = df['categories'].str.strip().str.lower().str.replace(' ','').str.split(';').str.join(' ')
df['popular_tags'] = df['popular_tags'].str.strip().str.lower().str.replace(' ','').str.split(',').str.join(' ')
df['languages'] = df['languages'].str.strip().str.lower().str.replace(' ','').str.split(',').str.join(' ')

#### These next two functions are helpful in removing any labels that only show up once as we want to focus on tags that are more common than that. 

In [17]:
def find_elements(column):
    all_elements = df[column].map(str).values.tolist()
    all_elements = ' '.join(all_elements)
    all_elements = all_elements.split()
    print('first elements len', len(all_elements))

    cnt = Counter()
    for word in all_elements:
        cnt[word] += 1
    print('first cnt len', len(cnt))

    for key, count in dropwhile(lambda key_count: key_count[1] >= 2, cnt.most_common()):
        del cnt[key]
    print('second cnt len', len(cnt))

    for i in range(0, len(all_elements)): 
        all_elements[i] = ''.join(all_elements[i]) 

    UniqW = Counter(all_elements) 

    all_elements = ' '.join(cnt.keys())
    all_elements = all_elements.split(' ')
    print('second elements len', len(all_elements))
    return cnt, all_elements

In [18]:
def remove_elements(column, all_elements): 
    df[column] = df[column].str.split(' ')
    for index, row in df.iterrows():
        current_elements = row[column]
        new_elements = []
        for element in current_elements:
            if element in all_elements:
                new_elements.append(element)
        df.at[index, column] = new_elements
    df[column] = df[column].str.join(' ')

In [19]:
lang_cnt, all_lang = find_elements('languages')

first elements len 64263
first cnt len 33
second cnt len 30
second elements len 30


In [20]:
remove_elements('languages', all_lang)

In [21]:
tags_cnt, all_tags = find_elements('popular_tags')

first elements len 150695
first cnt len 376
second cnt len 373
second elements len 373


In [22]:
# also drop earlyaccess, greatsoundtrack, masterpiece, kickstater
all_tags.remove('earlyaccess')
all_tags.remove('greatsoundtrack')
all_tags.remove('masterpiece')
all_tags.remove('kickstarter')

In [23]:
remove_elements('popular_tags', all_tags)

In [24]:
genres_cnt, all_genres = find_elements('genres')

first elements len 49428
first cnt len 27
second cnt len 26
second elements len 26


In [25]:
# lets also drop photoediting and 'massivelymultiplayer'
all_genres.remove('photoediting')
all_genres.remove('massivelymultiplayer')
all_genres.remove('earlyaccess')

In [26]:
remove_elements('genres', all_genres)

In [27]:
categories_cnt, all_categories = find_elements('categories')

first elements len 61367
first cnt len 27
second cnt len 27
second elements len 27


In [28]:
remove_elements('categories', all_categories)

#### Lets put all the text columns into one corpus for each game for a text vectorizer in the recommendation system.

In [29]:
text_cols = ['developer', 'publisher', 'platforms', 'categories', 'types', 'desc_snippet', 'popular_tags',
             'languages', 'mature_content']

df['soup'] = ''

for i in text_cols:
    df['soup'] += (' ' + df[i].map(str))

In [54]:
df.head()

name release_date         developer publisher  \
0            counterstrike   2000-11-01             valve     valve   
1    team fortress classic   1999-04-01             valve     valve   
2            day of defeat   2003-05-01             valve     valve   
3       deathmatch classic   2001-06-01             valve     valve   
4  halflife opposing force   1999-11-01  gearbox software     valve   

           platforms  required_age  \
0  windows mac linux             0   
1  windows mac linux             0   
2  windows mac linux             0   
3  windows mac linux             0   
4  windows mac linux             0   

                                          categories  genres  achievements  \
0  multi-player onlinemulti-player localmulti-pla...  action             0   
1  multi-player onlinemulti-player localmulti-pla...  action             0   
2                multi-player valveanti-cheatenabled  action             0   
3  multi-player onlinemulti-player localmulti-pla...  action             0   
4  single-player multi-player valveanti-cheatenabled  action             0   

     owners  ...  types                                       desc_snippet  \
0  15000000  ...    app  Play the world's number 1 online action game. ...   
1   7500000  ...    app  One of the most popular online action games of...   
2   7500000  ...    app  Enlist in an intense brand of Axis vs. Allied ...   
3   7500000  ...    app  Enjoy fast-paced multiplayer gaming with Death...   
4   7500000  ...    app  Return to the Black Mesa Research Facility as ...   

                                        popular_tags  \
0  action fps multiplayer shooter classic team-ba...   
1  action fps multiplayer classic shooter team-ba...   
2  fps worldwarii multiplayer shooter action war ...   
3  action fps classic multiplayer shooter first-p...   
4  fps action classic sci-fi singleplayer shooter...   

                                           languages  \
0  english french german italian spanish-spain si...   
1  english french german italian spanish-spain ko...   
2        english french german italian spanish-spain   
3  english french german italian spanish-spain ko...   
4                       english french german korean   

                                      mature_content  est_revenue  year  \
0  mature content description  the developers des...  107850000.0  2000   
1  mature content description  the developers des...   29925000.0  1999   
2                                               none   29925000.0  2003   
3                                               none   29925000.0  2001   
4                                               none   29925000.0  1999   

   month  season                                               soup  
0    nov    fall   valve valve windows mac linux multi-player on...  
1    apr  spring   valve valve windows mac linux multi-player on...  
2    may  spring   valve valve windows mac linux multi-player va...  
3    jun  summer   valve valve windows mac linux multi-player on...  
4    nov    fall   gearbox software valve windows mac linux sing...  

[5 rows x 21 columns]

#### Pickle data to be used in other notebooks
- whole_data to be used for the visulizations 
- simple_df to be used for the recommendations

In [53]:
df.to_pickle('data/whole_data.pkl')

In [32]:
simple_df  = df[['name', 'owners', 'price', 'est_revenue', 'season', 'year', 'soup']].copy()
simple_df.to_pickle('rec_app/simple.pkl')
simple_df.to_pickle('data/simple.pkl')